packages/lib-utils/src/k8s/k8s-resource.ts

```diff
@@ -0,0 +1,126 @@
+import type { AnyObject } from '@monorepo/common';
+import type { K8sModelCommon, K8sResourceCommon, QueryOptions, Patch } from '../types/k8s';
+import type { FetchOptionArgs } from '../utils/common-fetch';
+import { commonFetchJSON } from '../utils/common-fetch';
+import { getK8sResourceURL } from './k8s-utils';
+
+type K8sResourceReadArgs = [
+  model: K8sModelCommon,
+  options?: QueryOptions,
+  ...fetchOptions: FetchOptionArgs
+];
+
+type K8sResourceUpdateArgs<TResource extends K8sResourceCommon> = [
```

Comment:
Q: I assumed T stood for "Targeted" -- did you have another thought?

Reply:
`T` prefix for type parameters stands for "type" :smiley: another example would be `TValue` which means "value type"

```ts
export type CodeRef<TValue = unknown> = () => Promise<TValue>;
```

Reply:
Ah, dur lol -- thanks!